### PR TITLE
bug 1845173 - add ~nsCOMPtr and SegmentImpl to irrelevant list

### PR DESCRIPTION
**Pull request by amccreight**

### CODE REVIEW

File: socorro/signature/siglists/irrelevant_signature_re.txt [hunk 1]
```diff
@@ -162,6 +162,8 @@ mozilla::ThreadSafeAutoRefCnt
 mozilla::UniquePtr
 mozilla::widget::WlCrashHandler
 mozilla::wr::RenderThread::AddRef
+~nsCOMPtr
+~nsCOMPtr_base
```
Comment (willkg) [inline socorro/signature/siglists/irrelevant_signature_re.txt:166]: These show up in the signature like `nsCOMPtr_base::~nsCOMPtr_base`, so these two lines have no effect on signature generation.

Further, `nsCOMPtr` is in the prefix list so signature generation already covers those frames.

File: socorro/signature/siglists/irrelevant_signature_re.txt [hunk 2]
```diff
@@ -207,6 +209,7 @@ RustMozCrash
 _sigtramp
 _security_check_cookie
 __security_check_cookie
+SegmentImpl
```
Comment (willkg) [inline socorro/signature/siglists/irrelevant_signature_re.txt:212]: This shows up in the signature like `mozilla::SegmentedVector<T>::SegmentImpl<T>::~SegmentImpl`, so this line has no effect on signature generation.

However, you added `mozilla::SegmentedVector` to the prefix list in this PR. That does cause the signature to change. For example:

```
app@socorro:/app$ socorro-cmd signature 46954e0a-21c6-4e2b-b39b-2f5350230717
Crash id: 46954e0a-21c6-4e2b-b39b-2f5350230717
Original: nsCOMPtr<T>::~nsCOMPtr | mozilla::SegmentedVector<T>::SegmentImpl<T>::PopLast
New:      nsCOMPtr<T>::~nsCOMPtr | mozilla::SegmentedVector<T>::SegmentImpl<T>::PopLast | mozilla::SegmentedVector<T>::PopLastN | mozilla::dom::DeferredFinalizerImpl<T>::DeferredFinalize
Same?:    False
```
```

File: socorro/signature/siglists/irrelevant_signature_re.txt (hunk 1 above)
Comment (willkg) [inline socorro/signature/siglists/irrelevant_signature_re.txt:166]: These show up in the signature like `nsCOMPtr_base::~nsCOMPtr_base`, so these two lines have no effect on signature generation.

Further, `nsCOMPtr` is in the prefix list so signature generation already covers those frames.

**Review by willkg — CHANGES_REQUESTED** [review]
We could reduce the scope of this to just the commit adding `mozilla::SegmentedVector` to the prefix list and drop the other commit.

If you want to change signature generation to move some things to the irrelevant list, this will need changes.

File: socorro/signature/siglists/irrelevant_signature_re.txt (hunk 1 above)
Comment (amccreight) [reply socorro/signature/siglists/irrelevant_signature_re.txt:166]: Ah, yeah, I somehow managed to realize that the regular expression had to match from the start of the line. I'll update these.

File: socorro/signature/siglists/irrelevant_signature_re.txt (hunk 2 above)
Comment (amccreight) [reply socorro/signature/siglists/irrelevant_signature_re.txt]: I've update this in a new patch.

File: socorro/signature/siglists/irrelevant_signature_re.txt (hunk 1 above)
Comment (amccreight) [reply socorro/signature/siglists/irrelevant_signature_re.txt]: I've update this in a new patch. I don't know if there's a good way to rebase the two patches together from the GitHub UI or not.

Comment (willkg) [reply socorro/signature/siglists/irrelevant_signature_re.txt]: I can rebase it.

**Comment by willkg** [comment]
I rebased against main and squashed the nsCOMPtr commits.

Signature generation for the three crash reports in bug #1845173 now look like this:

```
app@socorro:/app$ socorro-cmd signature 46954e0a-21c6-4e2b-b39b-2f5350230717 be45dea0-52c0-4ce3-b4f6-98c920230723 c0a0387e-d40b-406f-bb44-7a34a0230719
Crash id: 46954e0a-21c6-4e2b-b39b-2f5350230717
Original: nsCOMPtr<T>::~nsCOMPtr | mozilla::SegmentedVector<T>::SegmentImpl<T>::PopLast
New:      mozilla::SegmentedVector<T>::PopLastN | mozilla::dom::DeferredFinalizerImpl<T>::DeferredFinalize
Same?:    False

Crash id: be45dea0-52c0-4ce3-b4f6-98c920230723
Original: nsCOMPtr<T>::~nsCOMPtr | mozilla::dom::SessionHistoryInfo::~SessionHistoryInfo
New:      mozilla::dom::SessionHistoryInfo::~SessionHistoryInfo
Same?:    False

Crash id: c0a0387e-d40b-406f-bb44-7a34a0230719
Original: nsCOMPtr_base::~nsCOMPtr_base | mozilla::SegmentedVector<T>::SegmentImpl<T>::PopLast
New:      mozilla::SegmentedVector<T>::PopLastN | mozilla::dom::DeferredFinalizerImpl<T>::DeferredFinalize
Same?:    False
```

@amccreight Does that look ok to you?

**Comment by amccreight** [comment]
Yes, that looks good. Thanks!

**Review by willkg — APPROVED** [review]

**Comment by willkg** [comment]
Thank you!

File: socorro/signature/siglists/irrelevant_signature_re.txt (hunk 1 above)
Comment (Mrvirt222) [reply socorro/signature/siglists/irrelevant_signature_re.txt]: Thanks

**Pull request merged**